ir/shared/src/main/scala/org/scalajs/ir/Trees.scala

```diff
@@ -322,7 +327,14 @@ object Trees {
     // String.length, introduced in 1.11
     final val String_length = 17
 
-    def resultTypeOf(op: Code): Type = (op: @switch) match {
+    // New in 1.17
+    final val CheckNotNull = 18
+
+    @deprecated("use the overload with the argument type instead", since = "1.17.0")
+    def resultTypeOf(op: Code): Type =
+      resultTypeOf(op, NothingType): Type
```

Comment:
I'd remove this. It doesn't yield a correct result for `CheckNotNull`.

Reply:
Done.